app/controllers/api/pdf_controller.rb

```diff
@@ -0,0 +1,45 @@
+class API::PdfController< ApplicationController
+  def show
+    render json: { errors: 'siren invalid' }, status: 400 and return unless Siren.new(siren).valid?
+    render json: { errors: 'siren not found' }, status: 404 and return if aucun_dossier
+    render json: { errors: 'data is corrupt or incomplete and cannot be served' }, status: 500 and return if cannot_serve
+
+    send_data pdf.render,
+              filename: "infos_#{siren}.pdf",
+              type: 'application/pdf'
+  end
+
+  private
+
+  def pdf
+    SirenInfosPdf.new(dossier_principal)
+  end
+
+  def siren
+    params.require(:siren)
+  end
+
+  def dossiers
+    @dossiers ||= DossierEntreprise.where(siren: siren)
+  end
+
+  def dossier_principal
+    @dossier_principal ||= dossiers.find_by(type_inscription: 'P')
+  end
+
+  def cannot_serve
+    dossier_principal_inexistant || trop_inscriptions_en_greffe_principal
+  end
+
+  def aucun_dossier
+    dossiers.empty?
+  end
+
+  def dossier_principal_inexistant
+    dossier_principal.nil?
+  end
+
+  def trop_inscriptions_en_greffe_principal
+    dossiers.where(type_inscription: 'P').count > 1
+  end
+end
```

Comment:
Dommage que ce soit pas en trailblazer pour la gestion du control flow. 
Une opération éliminerait les multiples if/else/unless du départ. De plus, ça permettrait d'avoir un message d'erreur plus explicite : des exemples de siren qui ne rentreront pas dans le cas d'usage il y en a beaucoup, et le message 'data is corrupt or incomplete and cannot be served' n'aide pas à identifier la raison pour laquel le siren ne rentre pas dans le périmètre.

Reply:
On fera ça plus tard, je voulais un truc minimal et j'ai repris le code de l'autre API.

Reply:
On peut pairer pour adapter en trailblazer la semaine prochaine, il reste du temps avant le 7 novembre, et je suis d'accord on peut faire d'une pierre deux coups et faire en sorte que l'autre API utilise le même code.

Reply:
ah ah ! j'avais aussi pensé à mettre ça dans le même controller, mais pour moi ça implique de le renommer, je comptais faire ça dans une autre PR du coup

Reply:
Nouvelle issue #19